app/components/polls/PollPage.tsx

```diff
@@ -0,0 +1,157 @@
+      <div className="flex flex-col relative items-stretch w-full h-full max-w-full mx-auto xl:w-1/2 xl:bg-zinc-900 xl:rounded-xl">
+        <div className="p-6 bg-zinc-800 xl:rounded-t-xl">
+          <div className="flex items-end justify-between ">
+            <p className="pl-2 mb-2 text-xs font-bold text-left uppercase border-l-2 text-zinc-400 border-l-zinc-400">
+              Topic
+            </p>
+
+            <div className="flex items-center gap-3 ml-auto mb-2">
+              <div className="text-white rounded-l text-xs">
+                <VisibilityIcon className="inline w-3 h-3 mr-1.5 mb-0.5" />
+
+                {visibilityText}
+              </div>
+
+              <div className="">
+                <QrCodeGenerator />
+              </div>
+
+              <Link href={`/polls/${poll?.slug}/results`} className="group">
+                <div className="flex items-center rounded-full bg-zinc-600 text-white text-xs px-3 py-[6px] group-hover:bg-zinc-500">
+                  <BarChartIcon className="inline w-3 h-3 mr-1.5" />
+                  Results
+                </div>
+              </Link>
+            </div>
+          </div>
+          <h1 className="font-semibold text-white">{poll?.title}</h1>
+          <h2 className="text-sm text-zinc-500">
+            {poll?.core_question ||
+              "What do you think of the following statements?"}
+          </h2>
+        </div>
+
+        {questionsRemaining ? (
+          <>
+            {statementsToHideIds && statements && (
+              <div className="px-6 pt-6">
+                <Progress
+                  className="w-full h-2"
+                  value={(statementsToHideIds.length / statements.length) * 100}
+                />
+              </div>
+            )}
+
+            {statementsWithoutResponsesAndFlags &&
+              statementsToHideIds &&
+              statementOptions && (
+                <Cards
+                  statements={statementsWithoutResponsesAndFlags}
+                  statementsToHideIds={statementsToHideIds}
+                  statementOptions={statementOptions}
+                />
+              )}
+
+            {poll && (
+              <div className="flex justify-center mt-8 mb-10 sm:mb-0 sm:mt-0 pb-8">
+                <CreateStatementButton pollId={poll.id} />
+              </div>
+            )}
+          </>
+        ) : (
+          children
+        )}
+      </div>
+      <Tutorial />
+    </main>
+  );
+}
+
+export const POLLED_POLL_QUERY_KEY = "/polls/[slug]";
+export function usePolledPollData(
+  initialData: Awaited<ReturnType<typeof getData>>,
+) {
+  const params = useParams();
+  const slug = params.slug;
+  const { data } = useQuery({
+    queryKey: [POLLED_POLL_QUERY_KEY, slug],
+    queryFn: async () => {
+      const res = await fetch(`/api/polls/${slug}`);
+      return res.json() as ReturnType<typeof getData>;
+    },
+    initialData,
+    refetchInterval: 15_000,
+  });
+
```

Comment:
Because we're passing `initialData` we know the data must be defined immediately on the client, so we can add something like `if (!data) notFound();` to help typescript realize the type will never be undefined.

By doing this you can remove all the `poll?.`, `statement?.` etc. through this file because they will be known.